app/src/main/java/com/simplecity/amp_library/ui/screens/album/detail/AlbumDetailFragment.kt

```diff
@@ -403,9 +403,12 @@ class AlbumDetailFragment :
 
     private var songClickListener: SongView.ClickListener = object : SongView.ClickListener {
         override fun onSongClick(position: Int, songView: SongView) {
-            val uiEvent = EventUtils.newUiEvent(songView.song, UiEventType.PLAY, context)
-            FirebaseIOUtils.saveUiEvent(uiEvent)
             if (!contextualToolbarHelper!!.handleClick(songView, Single.just(listOf(songView.song)))) {
+                val context = context
+                if (context != null) {
```

Comment:
Same - please remove null check

Reply:
When I remove the conditions, AS throws a `Typemismatch` error since `newUiEvent` takes in `context` instead of `context?`. However, it is not the same case for the `song` object.

Reply:
You can pass in `context!!` instead - that will throw an NPE if `context` is null.

However - let's use `ShuttleApplication.get()` instead of the `context` variable of the fragment. That will use the application context, which should always exist as long as the code is running, which is fine as the context is only used to get the string value from resources.